atc/worker/artifact_source.go

```diff
@@ -149,12 +164,33 @@ func (source *artifactSource) StreamTo(
 		err = source.p2pStreamTo(ctx, destination)
 	}
 
+	if err != nil {
+		return err
+	}
+
 	// Inc counter if no error occurred.
-	if err == nil {
-		metric.Metrics.VolumesStreamed.Inc()
+	metric.Metrics.VolumesStreamed.Inc()
+
+	usedResourceCache, found, err := source.resourceCacheFactory.FindResourceCacheByID(source.volume.GetResourceCacheID())
+	if err != nil {
+		logger.Error("artifactSource-StreamTo-failed-to-find-resource-cache", err)
+		return nil
 	}
+	if !found {
+		logger.Info("artifactSource.StreamTo-not-find-resource-cache, this should not happen",
+			lager.Data{"rcId": source.volume.GetResourceCacheID(), "volumeHandle": source.volume.Handle()})
+		return nil
```

Comment:
If this should not happen, we probably want to return a constructed error like `ErrStreamingResourceCacheNotFound`. Same reasoning as my comment above, where if this is not supposed to happen then there is something that went wrong either in the ATCs or the database.